core/src/main/java/hudson/model/RunParameterDefinition.java

```diff
@@ -164,20 +168,26 @@ public ParameterValue getDefaultParameterValue() {
         }
 
         Run<?,?> lastBuild = null;
+        Job project = getProject();
+
+        if (project == null) {
+            LOGGER.log(WARNING, "project should not be null when retrieving default parameter value");
```

Comment:
It should print the stacktrace IMHO. Otherwise it's hard to track it down.

E.g. 
```
if (project == null) {
  if (LOOGER.isLoggable(WARNING)) {
    LOGGER.log(WARNING, "project should not be null when retrieving default parameter value", new IllegalStateException());
  }
  return null;
}
```